### ISSUE TITLE
Dropdown: Not associated with label

### ISSUE THREAD

**Issue by sane6** [openvi-foundation/openvue#615]
### Describe the bug

The Dropdown component isn't properly associated with a label that has a `for` attribute matching the Dropdown's `input-id` prop. Clicking on the label doesn't focus the dropdown and the dropdown doesn't have an accessible name (it should be the labels text value, if properly associated).

I've observed the same issue with the Calendar, InputNumber, and InputSwitch components.

### Reproducer

https://codesandbox.io/s/nice-stallman-hf8kur

### PrimeVue version

3.26.0

### Vue version

3.x

### Language

TypeScript

### Build / Runtime

Vue CLI App

### Expected behavior

The dropdown should have a name in the accessibility tree that matches the associated label.
Clicking an associated label should focus the dropdown.

Copied to track in our fork; not affiliated with the original report being closed/resolved upstream

Originally reported at: [primefaces#2924](https://github.com/primefaces/primevue/issues/2924)
Originally reported by: https://github.com/zcdunn
